perftest/wallet.go

```
@@ -0,0 +1,425 @@
+func SecondsFromNowInSecs(seconds int64) int64 {
+	return time.Now().Unix() + seconds
+}
+
+func loginWallet(walletURL, userName, password string) (string, error) {
+	postBody, _ := json.Marshal(map[string]string{
+		"wallet":     userName,
+		"passphrase": password,
+	})
+	responseBody := bytes.NewBuffer(postBody)
+
+	URL := "http://" + walletURL + "/api/v1/auth/token"
+
+	resp, err := http.Post(URL, "application/json", responseBody)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+	sb := string(body)
+	if strings.Contains(sb, "error") {
+		return "", fmt.Errorf(sb)
+	}
+
+	// Get the token value out from the JSON
+	var result map[string]interface{}
+	json.Unmarshal([]byte(sb), &result)
+
+	return result["token"].(string), nil
+}
+
+func createWallet(walletURL, userName, password string) (string, error) {
+	postBody, _ := json.Marshal(map[string]string{
+		"wallet":     userName,
+		"passphrase": password,
+	})
+	responseBody := bytes.NewBuffer(postBody)
+
+	URL := "http://" + walletURL + "/api/v1/wallets"
+
+	resp, err := http.Post(URL, "application/json", responseBody)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+	sb := string(body)
+	if strings.Contains(sb, "\"error\"") {
+		return "", fmt.Errorf(sb)
+	}
+
+	// Get the token value out from the JSON
+	var result map[string]interface{}
+	json.Unmarshal([]byte(sb), &result)
+
+	return result["token"].(string), nil
+}
+
+func createKey(walletURL, userName, password, token string) (string, error) {
+	postBody, _ := json.Marshal(map[string]string{
+		"passphrase": password,
+	})
+	postBuffer := bytes.NewBuffer(postBody)
+
+	URL := "http://" + walletURL + "/api/v1/keys"
+
+	client := &http.Client{}
+	req, err := http.NewRequest(http.MethodPost, URL, postBuffer)
+	if err != nil {
+		log.Fatal(err)
```

maybe return the err?